lib/firenest/replicated_state/server.ex

```diff
@@ -0,0 +1,288 @@
+defmodule Firenest.ReplicatedState.Server do
+  @moduledoc false
+  use Firenest.SyncedServer
+
+  alias Firenest.SyncedServer
+  alias Firenest.ReplicatedState.{Store, Remote, Handler}
+
+  def child_spec({name, topology, handler, opts}) do
+    server_opts = [name: name, topology: topology]
+
+    %{
+      id: name,
+      start: {SyncedServer, :start_link, [__MODULE__, {name, handler, opts}, server_opts]}
```

Comment:
Same issue here. The modules key would be wrong. I would avoid redefining child_spec and instead focus on start_link.